### ISSUE TITLE
AttributeError("'list' object has no attribute 'is_Float'",)

### ISSUE THREAD

**Issue by goedman** [jlapeyre/Symata.jl#158]
Hi John,

Is below error a known issue?

  * ../examples/ch07/WRM/Ex.7.13.jl *
PyError ($(Expr(:escape, :(ccall(#= /Users/rob/.julia/packages/PyCall/rUul9/src/pyfncall.jl:44 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))) <class 'AttributeError'>
AttributeError("'list' object has no attribute 'is_Float'",)
  File "/Users/rob/.julia/packages/Conda/hsaaN/deps/usr/lib/python3.6/site-packages/sympy/core/function.py", line 787, in __new__
    obj = super(AppliedUndef, cls).__new__(cls, *args, **options)
  File "/Users/rob/.julia/packages/Conda/hsaaN/deps/usr/lib/python3.6/site-packages/sympy/core/function.py", line 443, in __new__
    pr2 = min(cls._should_evalf(a) for a in result.args)
  File "/Users/rob/.julia/packages/Conda/hsaaN/deps/usr/lib/python3.6/site-packages/sympy/core/function.py", line 443, in <genexpr>
    pr2 = min(cls._should_evalf(a) for a in result.args)
  File "/Users/rob/.julia/packages/Conda/hsaaN/deps/usr/lib/python3.6/site-packages/sympy/core/function.py", line 462, in _should_evalf
    if arg.is_Float:

NumericalMethodsforEngineers.jl: Error During Test at /Users/rob/.julia/dev/NumericalMethodsforEngineers/test/runtests.jl:56

Both Julia 1.0.1 and Julia 1.1 with Symata 0.4.5

Regards,
Rob

**Comment by jlapeyre** [comment]
Do you have any idea at which line in the example code the error is thrown ?

**Comment by goedman** [comment]
I have been experimenting with scripts like below. Occasionally it works and then other examples in ch07 work.

```
using NumericalMethodsforEngineers
println("\n\n--------1------------\n")

@sym begin
  ClearAll(xi, yi, Y, R, sol, ytilde)
  xi = [0, 1//2, 1]
  yi = [0, a, 1]
  Y(x_) := LagrangePolynomial(xi, yi)
  Println(Y(x))
end

println("\n---------2-----------\n")

@sym begin
  ClearAll(xi, yi, Y1, R, sol, ytilde)
  Y1(x_) := x^3 -3x^2 +6x - 9
  R(x_) = Simplify(D(Y1(x), x, 2))
  Println(R(x))
end

println("\n----------3----------\n")

@sym begin
  ClearAll(a, xi, yi, r, Y, R, sol, ytilde)
  xi = [0, 1//2, 1]
  yi = [0, a, 1]
  Y(x_) := LagrangePolynomial(xi, yi)
  R(x_) = Simplify(D(Y(x), x, 2) - 3*x - 4*Y(x))
  R(x_) = Simplify(R(x) ./ (a => -C1/4))
  Println(R(x))
  SetJ(r, ToString(Simplify(R(x))))
end

println("\n----------4----------\n")

println(r)

println("\n----------5----------\n")

println("\n\nExample 7.14: y'' = 3x + 4y, y(0)=0, y(1)=1")
println("Residual for Weighted Residual Method using 2 point Lagragian Polynomial")
@sym Println("\nY(x): ", Expand(Y(x)), "\n")
println("( Example 7.14 gives: ytilde = 1/2*x^3*(27a-27b+9)-x^2*(45a-36b+9)+x*(18a-9b+2) )")
@sym Println("\nR(x) = ", R(x), "\n")
```

I'm now experimenting with this in runtests.jl. I am a bit suspicious that in some cases it can't find the LagrangePolynomial function. This could be a consequence of the different testing paradigm.

**Comment by jlapeyre** [comment]
Yes, Please do leave it open. I don't have much time now to look at it.
